src/main/java/eu/dissco/backend/service/RabbitMqPublisherService.java

```diff
@@ -1,68 +1,94 @@
+    } catch (JsonProcessingException e) {
+      log.error(SERIALISATION_ERROR, event, e);
+      throw new ProcessingFailedException("Failed to serialize tombstone event", e);
+    }
   }
 
-  public void publishUpdateEvent(JsonNode object, JsonNode currentObject, Agent agent)
-      throws JsonProcessingException {
-    var event = provenanceService.generateUpdateEvent(object, currentObject, agent);
+  public void publishUpdateEvent(Object object, Object currentObject, Agent agent)
+      throws ProcessingFailedException {
+    var event = provenanceService.generateUpdateEvent(mapper.valueToTree(object),
+        mapper.valueToTree(currentObject), agent);
     log.info("Publishing new update message to queue: {}", event);
-    rabbitTemplate.convertAndSend(cutExchange, cutRoutingKey, mapper.writeValueAsString(event));
+    try {
+      rabbitTemplate.convertAndSend(rabbitMqProperties.getProvenanceExchange(),
+          assembleRoutingKey(object), mapper.writeValueAsString(event));
+    } catch (JsonProcessingException e) {
+      log.error(SERIALISATION_ERROR, event, e);
+      throw new ProcessingFailedException("Failed to serialize tombstone event", e);
+    }
   }
 
-  public void publishTombstoneEvent(JsonNode tombstoneObject, JsonNode currentObject, Agent agent)
-      throws JsonProcessingException {
-    var event = provenanceService.generateTombstoneEvent(tombstoneObject, currentObject, agent);
+  public void publishTombstoneEvent(Object tombstoneObject, Object currentObject, Agent agent)
+      throws ProcessingFailedException {
+    var event = provenanceService.generateTombstoneEvent(mapper.valueToTree(tombstoneObject),
+        mapper.valueToTree(currentObject), agent);
     log.info("Publishing new tombstone message to queue: {}", event);
-    rabbitTemplate.convertAndSend(cutExchange, cutRoutingKey, mapper.writeValueAsString(event));
+    try {
+      rabbitTemplate.convertAndSend(rabbitMqProperties.getProvenanceExchange(),
+          assembleRoutingKey(tombstoneObject), mapper.writeValueAsString(event));
+    } catch (JsonProcessingException e) {
+      log.error(SERIALISATION_ERROR, event, e);
+      throw new ProcessingFailedException("Failed to serialize tombstone event", e);
+    }
   }
 
   public void publishVirtualCollectionEvent(VirtualCollectionEvent event)
       throws JsonProcessingException {
     log.info("Publishing {} virtual-collection to queue", event.action());
-    rabbitTemplate.convertAndSend(virtualCollectionExchange, virtualCollectionRoutingKey,
+    rabbitTemplate.convertAndSend(rabbitMqProperties.getVirtualCollectionExchange(),
+        rabbitMqProperties.getVirtualCollectionRoutingKey(),
         mapper.writeValueAsString(event));
   }
 
+  private String assembleRoutingKey(Object object) throws ProcessingFailedException {
+    if (Objects.requireNonNull(object) instanceof VirtualCollection) {
+      return rabbitMqProperties.getProvenanceRoutingPrefix() + ".virtual-collection";
```

Comment:
i'd suggest you should include ".virtual-collection" in the provenance routing prefix property, but I assume the idea is to have multiple cases here eventually? such as the media derivative service?